trio/_tests/check_type_completeness.py

```diff
@@ -0,0 +1,159 @@
+        and current_result["generalDiagnostics"][0]["message"]
+        == "No py.typed file found"
+    ):
+        print("creating py.typed")
+        py_typed_file = (
+            Path(current_result["typeCompleteness"]["packageRootDirectory"])
+            / "py.typed"
+        )
+        py_typed_file.write_text("")
+
+        res = run_pyright()
+        current_result = json.loads(res.stdout)
+
+    if res.stderr:
+        print(res.stderr)
+
+    last_result = json.loads(RESULT_FILE.read_text())
+
+    for key in "errorCount", "warningCount", "informationCount":
+        check_zero(key, current_result["summary"])
+
+    for key, invert in (
+        ("missingFunctionDocStringCount", False),
+        ("missingClassDocStringCount", False),
+        ("missingDefaultParamCount", False),
+        ("completenessScore", True),
+    ):
+        check_less_than(
+            key,
+            current_result["typeCompleteness"],
+            last_result["typeCompleteness"],
+            invert=invert,
+        )
+
+    for key, invert in (
+        ("withUnknownType", False),
+        ("withAmbiguousType", False),
+        ("withKnownType", True),
+    ):
+        check_less_than(
+            key,
+            current_result["typeCompleteness"]["exportedSymbolCounts"],
+            last_result["typeCompleteness"]["exportedSymbolCounts"],
+            invert=invert,
+        )
+
+    assert (
+        res.returncode != 0
+    ), "Fully type complete! Delete this script and instead directly run `pyright --verifytypes=trio` (consider `--ignoreexternal`) in CI and checking exit code."
+
+    if args.overwrite_file:
+        print("Overwriting file")
+
+        # don't care about differences in time taken
+        del current_result["time"]
+        del current_result["summary"]["timeInSec"]
+
+        # don't fail on version diff so pyright updates can be automerged
+        del current_result["version"]
+
+        for key in (
+            # don't save path (because that varies between machines)
+            "moduleRootDirectory",
+            "packageRootDirectory",
+            "pyTypedPath",
+        ):
+            del current_result["typeCompleteness"][key]
+
+        # prune the symbols to only be the name of the symbols with
+        # errors, instead of saving a huge file.
+        new_symbols = []
+        for symbol in current_result["typeCompleteness"]["symbols"]:
+            if symbol["diagnostics"]:
+                new_symbols.append(symbol["name"])
+                continue
+
+        current_result["typeCompleteness"]["symbols"] = new_symbols
+
+        with open(RESULT_FILE, "w") as file:
+            json.dump(current_result, file, sort_keys=True, indent=2)
+            # add newline at end of file so it's easier to manually modify
+            file.write("\n")
+
+    if py_typed_file is not None:
+        print("deleting py.typed")
+        py_typed_file.unlink()
+
+    print("*" * 20)
+
+    return int(failed)
```

Comment:
`bool` is a subclass of `int` so this isn't necessary. But maybe nice for clarity? Dunno.

Reply:
haha, I guess it's my c++ programming leaking. But yeah I don't mind the int conversion